java/Oleksii-Riabov/src/com/zoolatech/lecture4/tasks/_1/Container.java

```diff
@@ -0,0 +1,29 @@
+package com.zoolatech.lecture4.tasks._1;
+
+public class Container {
+    String value;
+
+    public Container(String value) {
+        this.value = value;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public void setValue(String value) {
+        this.value = value;
+    }
+
+    public class ContainerUpdatesListener {
+        String innerValue;
+
+        public ContainerUpdatesListener() {
+            this.innerValue = value;
+        }
+
+        boolean hasValueChanged() {
```

Comment:
and the method can be public, since class is public and its the only functionality provide by the class

Reply:
Already made this method public.